File: honorary-one-line-wonders/javascript/ycombinator.js
var Fix = function (f) {
    return function (x) {
        return x(x);
    }(function (x) {
        return f(
            function (a) { return (x(x))(a); }
        );
    });
};


var fac = Fix(function(f){
    return function(n){
        if(n !== 0){
            return n * f(n-1);
        } else {
            return 1;
        }
    }
});

console.log(fac(2));
console.log(fac(8));

File: honorary-one-line-wonders/clojure/ycombinator.clj
(defn Fix [f]
  ((fn [x] (x x))
   (fn [x]
     (f (fn [& args]
          (apply (x x) args))))))

(def fac-rec
  (fn [f]
    (fn [n]
      if (zero? n) 1
        (* n (f (dec n)))))))


(def fac (Fix fac-rec))

(println (fac 2))
(println (fac 8))
